recipes/redboltz-mqtt_cpp/all/conanfile.py

```diff
@@ -1,64 +1,139 @@
-from conans import ConanFile, tools
-from conans.errors import ConanInvalidConfiguration
+from conan import ConanFile
+from conan.errors import ConanInvalidConfiguration
+from conan.tools.files import get, copy
+from conan.tools.cmake import CMake, CMakeDeps, CMakeToolchain
+from conan.tools.build import check_min_cppstd
+from conan.tools.scm import Version
 import os
 
-required_conan_version = ">=1.43.0"
+required_conan_version = ">=1.53.0"
 
-class MqttCPPConan(ConanFile):
+
+class MqttCppConan(ConanFile):
     name = "redboltz-mqtt_cpp"
     description = "MQTT client/server for C++14 based on Boost.Asio"
-    license = "BSL-1.0"
     url = "https://github.com/conan-io/conan-center-index"
     homepage = "https://github.com/redboltz/mqtt_cpp"
-    topics = ("mqtt", "boost", "asio")
-    settings = "os", "arch", "compiler", "build_type"
-    generators = "cmake"
+    topics = ("mqtt", "mqttv5", "tls", "boost", "websocket", "asio")
+    license = "BSL-1.0"
     no_copy_source = True
+    package_type = "header-library"
+    settings = "os", "arch", "compiler", "build_type"
+    options = {
+        "enable_std_containers": [True, False],
+        "with_tls": [True, False],
+        "use_logs": [True, False],
+        "use_websocket": [True, False],
+        "with_tuple_any_workaround": [True, False],
+        "mqtt_always_send_reason_code": [True, False],
+    }
+    default_options = {
+        "enable_std_containers":  False,
+        "with_tls":  False,
+        "use_websocket":  False,
+        "use_logs": True,
+        "with_tuple_any_workaround": False,
+        "mqtt_always_send_reason_code":  True,
+    }
+
+    def source(self):
+        get(self, **self.conan_data["sources"][self.version], strip_root=True)
 
     @property
-    def _source_subfolder(self):
-        return "source_subfolder"
+    def _min_cppstd(self):
+        if self.options.enable_std_containers:
+            return "17"
+        return "14"
+
+    @property
+    def _minimum_compiler_version(self):
+        return {
+            "17": {
+                "Visual Studio": "16",
```

Comment:
```suggestion
                "Visual Studio": "16",
                "msvc": "1923",
```
Conan 2.x does not use Visual Studio, but msvc only as compiler name.